src/model.js

```diff
@@ -2139,6 +2139,8 @@ Specify a different name for either index to resolve this issue.`);
 
     const countOptions = Utils.cloneDeep(options);
 
+    this._injectScope(options);
```

Comment:
This still mutates the `options` parameter, but more importantly, it doesn't seem like this solves the bug: The bug was related to the `count` query, but these options don't receive the scope anymore. Which indicates to me that the new test would pass even if we removed this line

Something else also concerns me: `findAll` already injects the scope. Injecting it here means it's injected twice for `findAll`, which will likely introduce bugs

Reply:
Yeah, I guess you got distracted once working on it but I had indeed reported that the `findAll` part of the `findAndCountAll` was injected correctly, only the `count` was bad. :)

At a glance, I suppose the meaning was that you should make a clone of the `options` and inject on this clone, instead of literally mutating/modifying the existing, original options, which could have side effects in other places and make debugging harder.

Regular `count` with scope works correctly I believe, maybe you can have a look at how this part does the scope injection as guideline?

Reply:
@laurelgr rip dumb me, I didn't got that (that's why you should always read properly first) lemme get onto this!